packages/backend/src/core/collectors/WithdrawalAllowedCollector.ts

```diff
@@ -70,17 +74,34 @@ export class WithdrawalAllowedCollector {
     }
 
     switch (event.name) {
-      case 'LogWithdrawalAllowed':
+      case 'LogWithdrawalAllowed': {
+        const freezeStatus = await this.keyValueStore.findByKey('freezeStatus')
+
+        const data = {
+          starkKey: StarkKey.from(event.args.starkKey),
+          assetType: AssetHash.from(event.args.assetType),
+          nonQuantizedAmount: event.args.nonQuantizedAmount.toBigInt(),
+          quantizedAmount: event.args.quantizedAmount.toBigInt(),
+        }
+
+        if (freezeStatus === 'frozen') {
```

Comment:
I guess this is a bit confusing, but freezeStatus is the status of the contract right now (it's set by calling `.isFrozen()` on the contract). So when explorer is resyncing, it might be processing LogWithdrawalEvents from the past, when the exchange was not frozen yet. So checking freezeStatus here is incorrect.

You can even imagine a scenario where freeze comes exactly after this event, so relying on anything that is set in a step which is separate to the collection process (e.g. before or after) could also return incorrect value here.

I think the best solution would be to add collector for an event that's fired when exchange is frozen (`emit LogFrozen()`)and here check if this event is in DB.